cmd/capibmadm/cmd/vpc/key/create.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package key
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	logf "sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/vpc"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+type keyCreateOptions struct {
+	name              string
+	publicKey         string
+	resourceGroupName string
+}
+
+// CreateCommand vpc key create command.
+func CreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create VPC key",
+		Example: `
+# Create key in VPC
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm vpc key create key-name --region <region> --resource-group-name <resource-group-name>
+ --public-key <public-key-string> `,
+	}
+
+	options.AddCommonFlags(cmd)
+	var keyCreateOption keyCreateOptions
+	cmd.Flags().StringVar(&keyCreateOption.resourceGroupName, "resource-group-name", keyCreateOption.resourceGroupName, "IBM cloud resource group name")
+	cmd.Flags().StringVar(&keyCreateOption.publicKey, "public-key", keyCreateOption.publicKey, "Public Key")
+	_ = cmd.MarkFlagRequired("public-key")
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if len(args) < 1 {
+			return fmt.Errorf("key-name name is not provided")
+		}
+
+		keyCreateOption.name = args[0]
+		if err := createKey(cmd.Context(), keyCreateOption); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func createKey(ctx context.Context, keyCreateOption keyCreateOptions) error {
+	log := logf.Log
+	v1, err := vpc.NewV1Client(options.GlobalOptions.VPCRegion)
```

Comment:
lets rename `v1` to something more descriptive.

Reply:
`v1` is rename to `vpcClient`